File: 3513.number-of-unique-xor-triplets-i.py
#
# @lc app=leetcode id=3513 lang=python3
#
# [3513] Number of Unique XOR Triplets I
#

# @lc code=start
class Solution:
    def uniqueXorTriplets(self, nums: List[int]) -> int:
        unique_xor_values = set()
        n = len(nums)
        for i in range(n):
            for j in range(i, n):
                for k in range(j, n):
                    xor_value = nums[i] ^ nums[j] ^ nums[k]
                    unique_xor_values.add(xor_value)
        return len(unique_xor_values)
# @lc code=end

File: 3513.number-of-unique-xor-triplets-i.cpp
# @lc app=leetcode id=3513 lang=cpp
#
# [3513] Number of Unique XOR Triplets I
#
# @lc code=start
class Solution {
public:
    int uniqueXorTriplets(vector<int>& nums) {
        // Since nums is a permutation of [1, n], all numbers from 0 to n-1 can be formed by XOR.
        return nums.size();
    }
};
# @lc code=end